cpp/locality/NeighborList.cc

```diff
@@ -212,14 +234,14 @@ bool compareNeighborDistance(const NeighborBond& left, const NeighborBond& right
 
 bool compareFirstNeighborPairs(const std::vector<NeighborBond>& left, const std::vector<NeighborBond>& right)
 {
-    if (left.size() && right.size())
+    if (!left.empty() && !right.empty())
     {
         return compareNeighborBond(left[0], right[0]);
     }
-    else
-    {
-        return left.size() < right.size();
-    }
+    // cppcheck erroneously flags this check as redundant because it follows
+    // both branches independently.
```

Comment:
Any further insight here? I'm not sure if I understand what you mean by "it follows both branches independently."

Reply:
If we reached this point, that means that the previous `if` statement was false, which means that one of the two lists was empty. `cppcheck` looks at the two possible cases independently: if `left` is empty then `left.size() < right.size()` is trivially true, and if `right` is empty then the statement is trivially false (for slightly different reasons, hence the two different warnings I'm suppressing). We could rewrite this line using more if statements to avoid the warnings, but I find the current idiom to be readable and easy to understand (not to mention simpler than any alternative).